File: src/java/org/javaturk/dp/ch03/ocp/factory/solution/ManagerFactory.java
package org.javaturk.dp.ch03.ocp.factory.solution;

import org.javaturk.dp.ch04.factoryMethod.employee.EmployeeRandomizer;

public class ManagerFactory implements Factory{

	@Override
	public Manager create() {
		Manager manager = new Manager(EmployeeRandomizer.createId(), EmployeeRandomizer.createName(),
				EmployeeRandomizer.createYear(), EmployeeRandomizer.createDepartment(), EmployeeRandomizer.createDepartment());
		return manager;
	}
}

File: src/c#/Creational/FactoryMethod/FactoryMethod2/ManagerFactory.cs
using System;
using Util;

namespace FactoryMethod2
{
    public class ManagerFactory : IFactory
    {
        public Employee Create(string name) { 
            string department = EmployeeRandomizer.CreateDepartment();
            Manager manager = new Manager(EmployeeRandomizer.CreateId(), name,
                EmployeeRandomizer.CreateYear(), department, department);
            return manager;
        }
    }
}
